src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -414,6 +424,87 @@ public Void clearEntitySet( UUID entitySetId ) {
         return null;
     }
 
+    @Override
+    @RequestMapping(
+            path = { "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + ENTITY_KEY_ID_PATH + "/" + NEIGHBORS },
+            method = RequestMethod.DELETE
+    )
+    public Void clearEntityAndNeighborEntities(
+            @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
+            @PathVariable( ENTITY_KEY_ID ) UUID entityKeyId
+    ) {
+
+        /*
+         * 1 - check permissions for the given EntitySet as well as all of its neighbor EntitySets
+         */
+
+        // getNeighborEntitySetIds() returns source, destination, and edge EntitySet ids
+        Set<UUID> neighborEntitySetIds = dgm.getNeighborEntitySetIds( entitySetId );
+        Set<UUID> allEntitySetIds = ImmutableSet.<UUID>builder()
```

Comment:
Just use neighborEntitySetIds.stream() and Stream.concat to avoid re-hashing every object in the entity set.

Reply:
These are UUIDs, not whole EntitySet objects. Is there really a major difference between `Stream.concat` and `.add` / `.addAll`?